common/network_manager.go

```diff
@@ -235,10 +237,19 @@ func (m *NetworkManager) handleRequestRedirect(req *Request, redirectResponse *n
 }
 
 func (m *NetworkManager) initDomains() error {
-	action := network.Enable()
-	if err := action.Do(cdp.WithExecutor(m.ctx, m.session)); err != nil {
-		return fmt.Errorf("unable to execute %T: %w", action, err)
+	actions := []Action{network.Enable()}
+
+	// Only enable the Fetch domain if necessary, as it has a performance overhead.
+	if m.userReqInterceptionEnabled {
+		actions = append(actions,
+			fetch.Enable().WithPatterns([]*fetch.RequestPattern{{URLPattern: "*"}}))
```

Comment:
I'm wondering if we should disable the network cache here as well, as it's done when setting up the [non-working](https://github.com/grafana/xk6-browser/issues/203) HTTP auth. See: https://github.com/grafana/xk6-browser/blob/ed0de4f92c91dcd848243861fd62a352bbbb1cc4/common/network_manager.go#L437-L443

@robingustafsson Can you comment on if this is required? I suppose it makes sense, just wanted to confirm.

Reply:
I'm not sure tbh. I've not found any clear info on how the `Network` and `Fetch` domains interact. I'd say it's a good precaution though to disable cache when we have to use `Fetch` domain given what we'll use it for.

Reply:
Done in 49b6724.